### PR TITLE
Fixed bugs related to express installer

### PR DESCRIPTION
**Pull request by utkarshgautamnyu**
With reference to the fogbugz case: https://highfidelity.fogbugz.com/f/cases/6834/Express-Installer-Install-express-over-existing-install-install-fails-previous-install-broken
Testing Notes:
1) Run the installer
2) Observe a new page - "Choose Installation Type"
a) "Express Install" is selected by default
b) Lower button panel has labels: Back, Install, Cancel
c) Click on "Custom Install"
d) Button panel changes to Back, Next, Cancel
3) Install with Express install selected
a) No Post Install Options page
b) Software installs with following defaults:
Create a desktop shortcut for Interface
Launch Sandbox after install
Launch Interface after install
Launch Sandbox on startup
4) Install with Custom Install selected
a) Post Install options page appears
b) If it is a PR build, Observe that checkbox for copying settings from Production is unchecked by default
c) Change any of the default settings (For example uncheck Desktop shortcut creation)
d) Observe the settings are reflected (For example Desktop shortcut is not created)
5)Come back and Install again with "Express Install" selected
a) Observe changes made in custom install (For example uncheck Desktop shortcut creation), are over ridden by defaults, i.e Desktop shortcut IS created
6)PR build tests:
a) In custom install, select the "copy from production" checkbox and install
b) Come back and install with Express Install, it **Does Not** copy settings/content from production

### CODE REVIEW

File: cmake/templates/NSIS.template.in [hunk 1]
```diff
@@ -638,15 +636,28 @@ Var LaunchServerNowState
...
+  ${If} $ExpressInstallState == ${BST_CHECKED}
+    StrCpy $Express "1"
+    
+    StrCpy $DesktopClientState ${BST_CHECKED}  
+    StrCpy $ServerStartupState ${BST_CHECKED} 
+    StrCpy $LaunchServerNowState ${BST_CHECKED}
+    StrCpy $LaunchClientNowState ${BST_CHECKED}
+
+    ${If} @PR_BUILD@ == 1
+      ${NSD_SetState} $CopyFromProductionCheckbox ${BST_UNCHECKED}
```
Comment (birarda) [inline cmake/templates/NSIS.template.in]: This should also use a StrCpy and the state

File: cmake/templates/NSIS.template.in [hunk 2]
```diff
@@ -853,8 +842,6 @@ Section "-Core installation"
   ; Rename the incorrectly cased Raleway font
   Rename "$INSTDIR\resources\qml\styles-uit\RalewaySemibold.qml" "$INSTDIR\resources\qml\styles-uit\RalewaySemiBold.qml"
 
-  ExecWait "$INSTDIR\vcredist_x64.exe /install /q /norestart"
```
Comment (birarda) [inline cmake/templates/NSIS.template.in]: This should not have been removed

File: cmake/templates/NSIS.template.in [hunk 3]
```diff
@@ -638,15 +636,28 @@ Var LaunchServerNowState
...
-    ${NSD_GetState} $CustomInstallRadioButton $CustomInstallState
+  ; check if the user asked for express/custom install
+  ${NSD_GetState} $ExpressInstallRadioButton $ExpressInstallState
+  ${NSD_GetState} $CustomInstallRadioButton $CustomInstallState
+  
+  ${If} $ExpressInstallState == ${BST_CHECKED}
+    StrCpy $Express "1"
+    
+    StrCpy $DesktopClientState ${BST_CHECKED}  
+    StrCpy $ServerStartupState ${BST_CHECKED}
```
Comment (birarda) [inline cmake/templates/NSIS.template.in]: We should also be setting the desktop server and clean install states unchecked here

**Comment by birarda** [comment]
Missing removal of check for `${If} ${SectionIsSelected} ${client}` in `InstallTypesPage` as well.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2327/HighFidelity-Beta-PR11137-dc4121f9d1bbdea78452859c82ae2fe3ee31fecf.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2327/HighFidelity-Beta-PR11137-dc4121f9d1bbdea78452859c82ae2fe3ee31fecf.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2331/HighFidelity-Beta-PR11137-56c695acbb1c132715c07810cf03b0fcc8040b18.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2331/HighFidelity-Beta-PR11137-56c695acbb1c132715c07810cf03b0fcc8040b18.exe

**Comment by cain-kilgore** [comment]
Testing.

**Comment by cain-kilgore** [comment]
# Test Failed
Test | Result
-------- | --------
Run the installer | Pass
Observe a new page - "Choose Installation Type" | Pass
a) "Express Install" is selected by default | Pass
b) Lower button panel has labels: Back, Install, Cancel | Pass
c) Click on "Custom Install" | Pass
d) Button panel changes to Back, Next, Cancel | Pass
Install with Express install selected | Pass
a) No Post Install Options page | Pass
b) Software installs with following defaults:
Create a desktop shortcut for Interface | Pass
Launch Sandbox after install | Pass
Launch Interface after install | Pass
Launch Sandbox on startup | Pass
Install with Custom Install selected | Pass
a) Post Install options page appears | Pass
b) If it is a PR build, Observe that checkbox for copying settings from Production is unchecked by default | Pass
c) Change any of the default settings (For example uncheck Desktop shortcut creation) | Pass
d) Observe the settings are reflected (For example Desktop shortcut is not created) | Pass
5)Come back and Install again with "Express Install" selected | Pass
a) Observe changes made in custom install (For example uncheck Desktop shortcut creation), are over ridden by defaults, i.e Desktop shortcut IS created | Failed (See notes)
6)**PR build tests**:
a) In custom install, select the "copy from production" checkbox and install | Pass
b) Come back and install with Express Install, it Does Not copy settings/content from production | Pass

## Test Notes

This was a strange one. When making modifications to the Custom Install options and installing High Fidelity, I re-launched the installer after it was complete and noticed that none of the options were ticked - even the ones I had manually ticked and unticked some others. They were all blank. Re-launching did not fix the issue. On first install the Custom Install options had some defaults ticked.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Pull request merged**